polkadot/xcm/pallet-xcm/src/tests/mod.rs

```diff
@@ -1760,3 +1774,346 @@ fn query_weight_to_asset_fee_noop() {
 		.unwrap());
 	})
 }
+
+#[test]
+fn test_xcm_send_precompile_works() {
+	use codec::Encode;
+
+	let balances = vec![
+		(ALICE, CUSTOM_INITIAL_BALANCE),
+		(ParaId::from(OTHER_PARA_ID).into_account_truncating(), CUSTOM_INITIAL_BALANCE),
+	];
+	new_test_ext_with_balances(balances).execute_with(|| {
+		let xcm_precompile_addr = H160::from(
+			hex::const_decode_to_array(b"00000000000000000000000000000000000A0000").unwrap(),
+		);
+
+		let sender: Location = AccountId32 { network: None, id: ALICE.into() }.into();
+		let message = Xcm(vec![
+			ReserveAssetDeposited((Parent, SEND_AMOUNT).into()),
+			ClearOrigin,
+			buy_execution((Parent, SEND_AMOUNT)),
+			DepositAsset { assets: AllCounted(1).into(), beneficiary: sender.clone() },
+		]);
+
+		let versioned_dest: VersionedLocation = RelayLocation::get().into();
+		let versioned_message: VersionedXcm<()> = VersionedXcm::from(message.clone());
+
+		let xcm_send_params = IXcm::xcmSendCall {
+			destination: versioned_dest.encode().into(),
+			message: versioned_message.encode().into(),
+		};
+		let call = IXcm::IXcmCalls::xcmSend(xcm_send_params);
+		let encoded_call = call.abi_encode();
+
+		let _ = pallet_revive::Pallet::<Test>::map_account(RuntimeOrigin::signed(ALICE));
+		let result = pallet_revive::Pallet::<Test>::bare_call(
+			RuntimeOrigin::signed(ALICE),
```

Comment:
Is there any way for the contract itself to call `send`? Triggering a `DescendOrigin` with the contract address?